### PR TITLE
Feature: use file models

### PR DESCRIPTION
**Pull request by Zorin95670**

### CODE REVIEW

File: src/parser/TerraformParser.js [hunk 1]
```diff
@@ -13,22 +13,23 @@ import Parser from 'src/antlr/terraformParser';
 class TerraformParser extends DefaultParser {
   /**
    * Indicate if this parser can parse this file.
-   * @param {String} [fileName] - Name of file.
+   * @param {FileInformation} [fileName] - Name of file.
```
Comment (k-y0u) [inline src/parser/TerraformParser.js:16]: All should be change, not only the type.

File: src/render/TerraformRenderer.js [hunk 2]
```diff
@@ -23,12 +24,12 @@ class TerraformRenderer extends DefaultRender {
   }
 
   /**
-   * Convert all provided components and links in terraform file content.
+   * Convert all provided components and links in terraform files.
    * @param {Component[]} components - List of components you want to convert.
    * @param {ComponentLink[]} links - List of links you want to convert.
-   * @return {String} - Generated file content from components and links.
+   * @return {FileInput[]} - Array of generated file content from components and links.
```
Comment (k-y0u) [inline src/render/TerraformRenderer.js:30]: `file content` -> `files`

File: src/render/TerraformRenderer.js [hunk 3]
```diff
@@ -23,12 +24,12 @@ class TerraformRenderer extends DefaultRender {
...
    */
-  render(componentsTree, links) {
+  render(componentsTree, links, defaultFileName = './new_file.tf') {
```
Comment (k-y0u) [inline src/render/TerraformRenderer.js:32]: missing description for the new param.

**Review by Hundraw — APPROVED** [review]

File: src/render/TerraformRenderer.js [hunk 4]
```diff
@@ -83,19 +86,54 @@ class TerraformRenderer extends DefaultRender {
...
+  }
+
+  /**
+   * Render files from related components.
+   * @param {Map<String,Component>} map - Component mapped by file name.
+   * @return {FileInput[]} Render files array.
+   */
+  generateFilesFromComponentsMap(map) {
+    const files = [];
+    map.forEach((components, path) => {
```
Comment (k-y0u) [inline src/render/TerraformRenderer.js:130]: `return Array.map()` should avoid 2 lines

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: src/parser/TerraformListener.js [hunk 5]
```diff
@@ -8,7 +8,7 @@ import {
 const getText = (ctx) => ctx.getText().replaceAll('"', '').trim();
 
 class TerraformListener extends antlr4.tree.ParseTreeListener {
-  constructor(definitions = []) {
+  constructor(definitions) {
```
Comment (k-y0u) [inline src/parser/TerraformListener.js:11]: no description for class and methods here

**Review by k-y0u — CHANGES_REQUESTED** [review]

**Review by k-y0u — APPROVED** [review]

**Pull request merged**